compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -1020,6 +1023,8 @@ protected Result seekMatch(ParserRuleContext currentCtx, int lookahead, int curr
                 case STRING_KEYWORD:
                     hasMatch = nextToken.kind == SyntaxKind.XML_KEYWORD;
                     break;
+                case NEW_EXPRESSION:
```

Comment:
Need to add `NEW_KEYWORD` to the `EXPRESSION_START` array. Otherwise, if theres an error before the new, expression, then error handler will not know to treat `new` as a expression.
i.e: Check something like: `T x  new();` where the equal sign is missing before new expression.

Reply:
fixed with 07c06fd